src/__test__/8bitChannelImages.write.test.ts

```diff
@@ -0,0 +1,62 @@
+import { createTestData } from './createTestData';
+import { testEncode } from './testEncode';
+
+describe('decode image with bitDepth of 1', () => {
+  it('decode an RGBA 2x2  image', () => {
+    // R G
+    // B W
+    const data = createTestData({
+      colorModel: 'RGBA',
+      width: 2,
+      height: 2,
+      data: new Uint8Array([
+        255, 0, 0, 255, 0, 255, 0, 255, 0, 0, 255, 255, 255, 255, 255, 255,
+      ]),
+    });
+    testEncode(data, '2x2RGBA.bmp');
```

Comment:
Unless I am missing something, this is not in line with what you commented. I think the `compression` value is `0` here.

> After discussion, compression and colorMasks were left as fields. BI_BITFIELDS compression is necessary to decode/encode RGBA images, since BI_RGB compression does not support alpha channel. Color masks are left to align with the way GIMP encodes images.

Reply:
Yes, this should bi fixed. With `compression:0` this image becomes `RGBX` image instead of `RGBA`.